core/src/main/java/io/appflate/restmock/utils/RequestMatchers.java

```diff
@@ -66,6 +72,55 @@ protected boolean matchesSafely(RecordedRequest item) {
         };
     }
 
+    public static RequestMatcher hasQueryParameters() {
+      return new RequestMatcher("matched query parameters") {
+        @Override
+        protected boolean matchesSafely(RecordedRequest item) {
+          try {
+            URL mockUrl = new URL("http", "localhost", item.getPath());
+            Map<String, List<String>> queryParams = RestMockUtils.splitQuery(mockUrl);
+            return queryParams.size() > 0;
+          } catch (MalformedURLException e) {
+            return false;
+          } catch (UnsupportedEncodingException e) {
+            return false;
+          }
+        }
+      };
+    }
+
+    public static RequestMatcher hasQueryParameters(final SimpleEntry<String, String>... expectedParams) {
```

Comment:
I'd preffer here a dedidacted `QueryParam` class instead of `SimpleEntry` as so the users won't have problems as of what the parameter is all about.